src/App.jsx

```diff
@@ -14,45 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_INDEX_DESC = 'reverse default';
+const ASC = 'asc';
+const DESC = 'desc';
+
+function getSortGoods(goods, sortField, sortDirection) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_BY_ALPHABET:
+          return sortDirection === ASC
+            ? good1.localeCompare(good2)
+            : good2.localeCompare(good1);
+
+        case SORT_BY_LENGTH:
+          // eslint-disable-next-line no-nested-ternary
+          return sortDirection === ASC
+            ? good1.length - good2.length
+            : (good1.length === good2.length)
+              ? good2.localeCompare(good1)
+              : good2.length - good1.length;
+
+        case SORT_BY_INDEX_DESC:
+          return preparedGoods.indexOf(good2) - preparedGoods.indexOf(good1);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [sortDirection, setSortDirection] = useState(ASC);
+  const sortGoods = getSortGoods(goodsFromServer, sortField, sortDirection);
+  const sortReverse = () => {
+    if (sortField === '') {
```

Comment:
Instead of checking if the sortField is an empty string, it would be better to check if it's falsy. This will also take into account null and undefined values.

Reply:
Can`t understand how to solve this. Change '' on false, but it is not working

Reply:
If you have any questions, please ask them in fe_chat and tag mentors. It will be faster, because your answers in GitHub will be visible to mentor only after you re-request review
```suggestion
    if (!sortField) {
```

Reply:
Replied to you in the comment below with a request for correction